File: math/code/stern-brocot/fraction-finding-1.cpp
#include <iostream>
#include <string>

// Locate a given fraction in the Stern-Brocot tree.
std::string find(int x, int y, int a = 0, int b = 1, int c = 1, int d = 0) {
  int m = a + c, n = b + d;
  if (x == m && y == n) return "";
  if (x * n < y * m) {
    return 'L' + find(x, y, a, b, m, n);
  } else {
    return 'R' + find(x, y, m, n, c, d);
  }
}

int main() {
  int x, y;
  std::cin >> x >> y;
  std::cout << find(x, y);
  return 0;
}

File: math/code/stern-brocot/fraction-finding-1.py
# STERN-BROCOT TREE FRACTION FINDING.
# --8<-- [start:core]
# Locate a given fraction in the Stern-Brocot tree.
def find(x, y, a=0, b=1, c=1, d=0):
    m = a + c
    n = b + d
    if x == m and y == n:
        return ""
    if x * n < y * m:
        return "L" + find(x, y, a, b, m, n)
    else:
        return "R" + find(x, y, m, n, c, d)


# --8<-- [end:core]
if __name__ == "__main__":
    x, y = map(int, input().split())
    print(find(x, y))
